db-migration/db/migrate/20160609164952_create_name_string_indices.rb

```diff
@@ -0,0 +1,11 @@
+class CreateNameStringIndices < ActiveRecord::Migration
+  def change
+    create_table :name_string_indices, id: false do |t|
+      t.integer :data_source_id, null: false
+      t.uuid :name_string_id, null: false
+      t.string :url, limit: 255
+    end
+    add_index :name_string_indices, :data_source_id, using: 'btree'
```

Comment:
in MySQL world Percona makes some new developments with Fractal Trees https://en.wikipedia.org/wiki/TokuDB